src/modules/launcher/Plugins/Microsoft.Plugin.Program/Programs/UWPApplication.cs

```diff
@@ -394,72 +378,277 @@ internal string LogoPathFromUri(string uri, string theme)
+            return false;
+        }
+
+        internal void LogoPathFromUri(string uri, Theme theme)
+        {
+            // all https://msdn.microsoft.com/windows/uwp/controls-and-patterns/tiles-and-notifications-app-assets
+            // windows 10 https://msdn.microsoft.com/en-us/library/windows/apps/dn934817.aspx
+            // windows 8.1 https://msdn.microsoft.com/en-us/library/windows/apps/hh965372.aspx#target_size
+            // windows 8 https://msdn.microsoft.com/en-us/library/windows/apps/br211475.aspx
+            string path;
+            bool isLogoUriSet;
+            if (uri.Contains("\\", StringComparison.Ordinal))
+            {
+                path = Path.Combine(Package.Location, uri);
+            }
+            else
+            {
+                // for C:\Windows\MiracastView etc
+                path = Path.Combine(Package.Location, "Assets", uri);
+            }
+
+            if (theme == Theme.HighContrastBlack || theme == Theme.HighContrastOne || theme == Theme.HighContrastTwo)
+            {
+                isLogoUriSet = SetHighContrastIcon(path, ContrastBlack);
+            }
+            else if (theme == Theme.HighContrastWhite)
+            {
+                isLogoUriSet = SetHighContrastIcon(path, ContrastWhite);
+            }
+            else if (theme == Theme.Light)
+            {
+                isLogoUriSet = SetColoredIcon(path, ContrastWhite);
+            }
+            else
+            {
+                isLogoUriSet = SetColoredIcon(path, ContrastBlack);
+            }
+
+            if (!isLogoUriSet)
+            {
+                LogoPath = string.Empty;
+                LogoType = LogoType.Error;
+                ProgramLogger.LogException(
+                            $"|UWP|LogoPathFromUri|{Package.Location}" +
+                            $"|{UserModelId} can't find logo uri for {uri} in package location: {Package.Location}", new FileNotFoundException());
+            }
+        }
+
         public ImageSource Logo()
         {
-            var logo = ImageFromPath(LogoPath);
-            return logo;
+            if (LogoType == LogoType.Colored)
+            {
+                var logo = ImageFromPath(LogoPath);
+                var platedImage = PlatedImage(logo);
+                return platedImage;
+            }
+            else
+            {
+                return ImageFromPath(LogoPath);
+            }
+        }
+
+        private ImageSource PlatedImage(BitmapImage image)
+        {
+            if (!string.IsNullOrEmpty(BackgroundColor))
+            {
+                string currentBackgroundColor;
+                if (BackgroundColor == "transparent")
+                {
+                    currentBackgroundColor = SystemParameters.WindowGlassBrush.ToString(CultureInfo.InvariantCulture);
+                }
+                else
+                {
+                    currentBackgroundColor = BackgroundColor;
+                }
+
+                var padding = 8;
```

Comment:
(Clarification) Would this value not depend on the monitor properties?

Reply:
WPF application are written in monitor independent units. For example, [height](https://docs.microsoft.com/en-us/dotnet/api/system.windows.frameworkelement.actualheight?view=netcore-3.1#System_Windows_FrameworkElement_ActualHeight) for image control is set with device-independent units. You can find a detailed explanation here : https://www.codeproject.com/Articles/43520/Device-Independent-Units-to-Points-to-Inches